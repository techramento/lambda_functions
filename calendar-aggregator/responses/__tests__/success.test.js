const successResponse = require('../success')

const MockData = [
  {
    item: 'mock'
  }
]
const MockInput = Object.freeze({})

describe('success responses', () => {
  it('calls the callback with the calendar data', () => {
    successResponse(MockInput, (err, response) => {
      expect(err).toBeNull()
      expect(response).toMatchSnapshot()
    })(MockData)
  })

  it('does not crash if the callback itself throws', () => {
    const exception = new Error('MOCK EXCEPTION')
    const origConsole = console.error
    console.error = jest.fn()
    successResponse(MockInput, () => {
      throw exception
    })(MockData)
    expect(console.error).toHaveBeenCalledWith(exception)
    console.error = origConsole
  })
})
