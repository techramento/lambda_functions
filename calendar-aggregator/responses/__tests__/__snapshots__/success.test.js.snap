// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`success responses calls the callback with the calendar data 1`] = `
Object {
  "body": "{\\"data\\":[{\\"item\\":\\"mock\\"}],\\"input\\":{}}",
  "statusCode": 200,
}
`;
